### PR TITLE
Adds HTTP Data Structure

### PR DESCRIPTION
**Pull request by imdominicreed**
We need data structure to organize the HTTP data. This adds it for POST
and GET since these are the main two that we will currently be using.

### CODE REVIEW

File: http.hh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+struct http_get : http_request {
+  std::string url;
+  std::unordered_map<std::string, std::string> args;
+};
+
+struct http_post : http_request {
+  std::string url;
+  int fd;
```
Comment (evanugarte) [inline http.hh]: what is fd? can we give it a better name?

File: http.hh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#include <unordered_map>
+
+enum http_type { get, post };
+
+struct http_request {
+  http_type type;
+};
+
+struct http_get : http_request {
+  std::string url;
```
Comment (evanugarte) [inline http.hh]: `url` is in both structs lets move it to `http_request`

File: http.hh (hunk 1 above)
Comment (imdominicreed) [reply http.hh]: its the file descriptor where we will store the binary data.

Comment (imdominicreed) [reply http.hh]: ill change it to file_descriptor

Comment (evanugarte) [reply http.hh]: wouldn't that be generated on the server side? we might not need this here and can just use the `http_request` struct for both get and post

Comment (imdominicreed) [reply http.hh]: i was thinking that the serverside parses the requests into these data structures and passes them to the handlers.

Comment (imdominicreed) [reply http.hh]: then we can have the lambdas that take in the http request as a parameter and returns some http response.

File: http.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#include <string>
+#include <unordered_map>
+
```
Comment (evanugarte) [inline http.h:3]: this should probably live inside of a namespace if were gonna follow the google style guide (see this section on [namespaces](https://google.github.io/styleguide/cppguide.html#Namespaces))

we could do something like

```cpp
namespace domino {
namespace http {
  enum Method { kGet, kPost };
  struct request {
    Method method;
    // if the url was http://localhost:3000/upload?param=1
    // the endpoint should just be "upload"
    std::string endpoint;
  };
} // namespace http
} // namespace domino
```

then things can get super clean like

```cpp
domino::http::method::kGet;
domino::http::method::kPost;

domino::http::request request;

request.method
request.endpoint;
```

**Pull request merged**